src/common/walFilter/walFilter.c

```diff
@@ -33,33 +34,22 @@ typedef enum
 
 typedef struct WalInterface
 {
-    unsigned int pgVersion;
-    StringId fork;
-
-    uint16_t header_magic;
-    void (*validXLogRecordHeader)(const XLogRecord *record, PgPageSize heapPageSize);
-    void (*validXLogRecord)(const XLogRecord *record, PgPageSize heapPageSize);
-    pg_crc32 (*xLogRecordChecksum)(const XLogRecord *record, PgPageSize heapPageSize);
+    uint16 header_magic;
+    void (*validXLogRecordHeader)(const XLogRecordBase *record, PgPageSize heapPageSize);
+    void (*validXLogRecord)(const XLogRecordBase *record, PgPageSize heapPageSize);
+    uint32 (*xLogRecordHeaderSize)(void);
+    uint32 (*xLogRecordRmidSize)(void);
```

Comment:
These sizes are constants. I suggest to use variables like header_magic instead of pointers to function